bundles/binding/org.openhab.binding.serial/src/main/java/org/openhab/binding/serial/internal/RegexPatternMatcher.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.serial.internal;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.openhab.core.transform.TransformationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Pattern matcher to match and cache patterns
+ *
+ * @author Marek Halmo
+ * @see RegexPatternMatcher#getMatches getMatches()
+ *
+ */
+public class RegexPatternMatcher {
+    private static final Logger logger = LoggerFactory.getLogger(RegexPatternMatcher.class);
+    private static HashMap<String, Pattern> patternCache = new HashMap<String, Pattern>();
+    private static final Pattern SUBST_PATTERN = Pattern.compile("^s/(.*?[^\\\\])/(.*?[^\\\\])/(.*)$");
+
+    /**
+     * Returns compiled pattern for specific regeExpression that is stored in patternCache
+     *
+     * @param regExpression
+     * @return
+     */
+    private static Pattern getPattern(String regExpression) {
+        cache(regExpression);
+        return patternCache.get(regExpression);
+    }
+
+    /**
+     * Returns true if given regExpression is a substitution pattern
+     * This fact is indicated by giving
+     *
+     * @param regExpression
+     * @return
+     */
+    private static boolean isSubtitution(String regExpression) {
+        cache(regExpression);
+        return patternCache.get(regExpression) == null;
+    }
+
+    /**
+     * Caches given regexpression, if the expression is substitution pattern it marks is as substitution
```

Comment:
"Caches given regExpression. If the expression is a substitution pattern, it marks it as a substitution."